### ISSUE TITLE
Clipping the bar outside of the grid

### ISSUE THREAD

**Issue by Remy-J** [apache/echarts#13321]
### Version
4.6.0

### Reproduction link

[https://gallery.echartsjs.com/editor.html?c=xs9BT4SnHR&v=1](https://gallery.echartsjs.com/editor.html?c=xs9BT4SnHR&v=1)

### Steps to reproduce
use this conf: 
```js
option = {
    legend: {},
    tooltip: {},
    dataset: {
        source: [
            ['val', 'id'],
            [0, 1234],
            [4, 12345]
        ]
    },
    xAxis: {type: 'value'},
    yAxis: {},
    series: [
        {type: 'bar'}
    ]
};
```

### What is expected?
the graph is expected to be consistent

### What is actually happening?
the layout is broken.
the columns come out of the graph and their size is not good

**Comment by pissang** [comment]
You can specify the width of bar

**Comment by Remy-J** [comment]
thanks for your feedback but even by reducing the width of the bars when zooming the layout is broken

use this cofig: option = {
  dataZoom: [{ type: 'inside' }],
  grid: { bottom: 40, left: 70, right: 50, top: 30 },
  series: [
    {
      barWidth: '10%',
      data: [
        ['val', 'id'],
        [0, 377146],
        [1, 620432],
      ],
      name: 'id',
      type: 'bar',
    },
  ],
  toolbox: {
    feature: {
      dataZoom: { title: { back: 'Zoom Reset', zoom: 'Zoom' } },
    },
    magicType: {
      show: false,
      title: {
        bar: 'Switch to Bar Chart',
        line: 'Switch to Line Chart',
        stack: 'Stack',
        tiled: 'Tile',
      },
      type: [],
    },

    saveAsImage: { title: 'Save image' },

    show: false,
  },
  tooltip: { axisPointer: { type: 'shadow' }, trigger: 'axis' },

  xAxis: {
    axisLabel: {
      rotate: 0,
    },
    axisLine: {
      lineStyle: {
        color: '#656565',
      },
    },

    boundaryGap: ['20%', '20%'],
    max: 2,
    min: -1,
    scale: true,

    splitLine: {
      lineStyle: {
        color: '#DDD',
        type: 'dashed',
      },
    },

    type: 'value',
  },
  yAxis: {
    axisLine: {
      onZero: false,
    },
    scale: false,

    splitLine: {
      lineStyle: { color: '#DDD', type: 'dashed' },
    },
  },
};

here is a link where you can test:
https://gallery.echartsjs.com/editor.html?c=xs9BT4SnHR&v=3

place your mouse to the right of the first bar and zoom step by step and you will see the outdated bar of the y-axis

here is a link where you can see the result:
https://zupimages.net/viewer.php?id=20/40/vs3s.png

**Comment by Ovilia** [comment]
Hi, this is the expected behavior for now. Bars will be only clipped (display nothing) when all of the bar are outside of the grid area. There's no way to clip the part outside of the grid for now but it may be supported afterwards.

**Comment by Remy-J** [comment]
Hi, thank you for your reply

**Comment by Remy-J** [comment]
hello I would like to know if you have been able to make any progress on this issue.

**Comment by 100pah** [comment]
Fix in v6.1.0
